include/zephyr/drivers/watchdog/wdt_xilinx_axi.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright © 2023 Calian Ltd.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_WATCHDOG_WDT_XILINX_AXI_H_
+#define ZEPHYR_INCLUDE_DRIVERS_WATCHDOG_WDT_XILINX_AXI_H_
+
+#include <zephyr/device.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Get whether the Xilinx AXI watchdog triggered the last system reset.
+ *
+ * @param dev Watchdog device instance.
+ * @param last_watchdog_reset Set to indicate whether watchdog triggered last reset.
+ *
+ * @retval 0 If successful.
+ * @retval -errno In case of any other failure.
+ */
+int wdt_xilinx_axi_get_last_watchdog_reset(const struct device *dev, bool *last_watchdog_reset);
+
```

Comment:
Do you want to introduce new watchdog API to check a reason of reset?

Reply:
Yes, it would be possible to add this as a generic API. Does the way this call is set up (just returning a bool to indicate if the WDT believes it triggered the last system reset) make sense? I suppose that piece of information should be fairly generic to many watchdog devices - getting the exact reset reason may be outside of the scope of the watchdog hardware in some cases.

Reply:
I agree with you. It's useful to get the reason of previous reset.
Anyway, this topic is out of scope of this PR. Thanks for your comments.

Reply:
The driver is now optionally implementing the HWINFO API to get whether the last system reset was caused by the WDT.